infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -49,24 +45,48 @@ public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
      * @param r   The router that took in the message.
      */
     public void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        if (shutdown) {
+        if (getState() == ServerState.SHUTDOWN) {
             log.warn("Server received {} but is already shutdown.", msg.getMsgType().toString());
             return;
         }
 
+        if (!isServerReadyToHandleMsg(msg)) {
```

Comment:
Why is this a better place for checking the Not_ready compared to the check was being done in the CorfuMsgHandler?

Reply:
Because this is an internal state of AbstractServer. The server in charge of this check. If CorfuMsgHandler checks this we violate Single Responsibility principle